File: csharp/two-fer/TwoFer.cs
using System;

public static class TwoFer
{
    private const string MessageTemplate = "One for {0}, one for me.";
    private const string DefaultName = "you";

    public static string Speak()
    {
        return Speak(String.Empty);
    }

    public static string Speak(string name)
    {
        name = name?.Trim();
        if (String.IsNullOrEmpty(name))
        {
            name = DefaultName;
        }
        return String.Format(MessageTemplate, name);
    }
}

File: cpp/two-fer/two_fer.cpp
#include "two_fer.h"
#include <string>

namespace two_fer
{
	std::string two_fer(std::string your_name)
	{
		if (your_name.empty()) {
			your_name = "you";
		}

		return "One for " + your_name + ", one for me.";
	}
} // namespace two_fer

